software/bootcamp-code/main.py

```diff
@@ -3,17 +3,91 @@
 import motion
 import cv2
 import time
-from enum import Enum
+from enum import Enum # for state machine enumerate
+import asyncio #for referee commands over websockets
+import websockets #before use: pip3.9 install websockets
+import json #for parsing referee commands into python library
 
-# STATE MACHINE: 3 states - FIND_BALL = 0, CENTER_BALL = 1, DRIVE_TO_BALL = 2
+# method for listening to referee commands
+async def listen_referee(command_list):
+    async with websockets.connect('ws://192.168.3.26:8222') as websocket:
+        command = await websocket.recv()
+        command_list.append(command) #adds to the end of the list
+
+#STATE MACHINE
 class State(Enum):
     FIND_BALL = 0
-    CENTER_BALL = 1
-    DRIVE_TO_BALL = 2
+    MOVE_CENTER_BALL = 1
+    FIND_BASKET = 2
+    THROW_BALL = 3
+    STOP = 4
+
+
+global state
+global basket_blue
+
+# function to get the latest referee command
+def get_referee_commands(command_list):
+    # parse referee commands into python library (https://www.w3schools.com/python/python_json.asp)
+    referee=json.loads(command_list[-1])
+    command=referee["signal"]
+    if command=="start" and ("twelve" in referee["targets"]):
```

Comment:
The name of the robot should be a separate variable. It should be easy to change when the organizers assign their own ID-s to your robot.